plugins/pencilblue/controllers/actions/admin/sites/activate_site.js

```diff
@@ -0,0 +1,45 @@
+/*
+ Copyright (C) 2015  PencilBlue, LLC
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+module.exports = function(pb) {
+    
+    //pb dependencies
+    var util = pb.util;
+
+    function ActivateSite(){}
+    util.inherits(ActivateSite, pb.BaseController);
+
+    ActivateSite.prototype.render = function(cb)
+    {
+		var self = this;
+        var vars = this.pathVars;
+
+        var message = this.hasRequiredParams(vars, ['id']);
+        if(message) {
+            self.formError(message, '/admin/sites', cb);
+            return;
+        }
+
+        var siteService = new pb.SiteService();
+        var jobId = siteService.activateSite(vars.id);
+        var content = pb.BaseController.apiResponse(pb.BaseController.API_SUCCESS, '', jobId);
+        cb({content: content});
+    };
```

Comment:
I think this controller mixes two concepts.  The error checking calls upon a function that deals with forms but the happy path calls back with an API response.  

Is this a view controller or API controller?